Prettier Standard formatting

eddiewang opened the pull request:
This should be merged before #794 so I can rebase that branch with master. Thanks to @MSevey for the idea.

This PR is the simply the result of running `prettier-standard 'js/**/*.js' 'plugins/**/*.js'` in the repo.

Standard JS (https://github.com/standard/standard) is a popular style guide. I've used quite a few (Airbnb, Github, Google) and I always find myself coming back to Standard.

Prettier (https://github.com/prettier/prettier) is an auto-formatting tool that keeps code style consistent.

eddiewang committed:
Prettier standardjs formatting

lukechampine commented on js/rendererjs/disabledplugin.js:
```diff
@@ -28,10 +28,15 @@ const handleReport = () => {
+			<a style={reportStyle} onClick={handleReport}>
+				here.
+			</a>
+		</h2>
```
hmm. Does this affect the actual display of this element? I feel like a JS formatter shouldn't be touching HTML.

eddiewang replied:
note that this is jsx, not html. although it looks similar, there are some differences to how it works. It wouldn't affect the actual display of the element because jsx elements transpiles into javascript objects.

lukechampine commented on plugins/Files/js/components/deletedialog.js:
```diff
@@ -13,7 +13,8 @@ const DeleteDialog = ({files, actions}) => {
 				<div className="delete-text">
-					Are you sure you want to delete {files.size} {files.size === 1 ? ' file' : ' files'}
+					Are you sure you want to delete {files.size}{' '}
+					{files.size === 1 ? ' file' : ' files'}
```
the `{' '}` looks wrong to me, given that the following strings are also prefixed by a space.

eddiewang replied:
{' '} represents a space. It's somewhat verbose but generally clearer to read because we're writing a mixture of JS and HTML in these files.

lukechampine replied:
Sure, but in a lot of these cases we don't really care about the space. I think we often did stuff like:
```html
<h1> Some text here </h1>
```
to separate the content from the tags, but the formatter thinks that we want to render that extra space -- we don't.

lukechampine commented on plugins/Files/js/components/directoryinfobar.js:
```diff
@@ -20,13 +25,20 @@ const DirectoryInfoBar = ({path, nfiles, onBackClick, setDragFolderTarget}) => {
+				<span style={{ marginRight: '10px' }}> {path} </span>
+				<span style={{ marginRight: '10px' }}>
+					{' '}
+					{nfiles} {nfiles === 1 ? 'file' : 'files'}
```
ditto

lukechampine commented on plugins/Files/js/components/unlockwarning.js:
```diff
@@ -1,11 +1,16 @@
-		<h1 className="unlock-warning-head"> Your wallet must be unlocked and synchronized to buy storage. </h1>
+		<h1 className="unlock-warning-head">
+			{' '}
+			Your wallet must be unlocked and synchronized to buy storage.{' '}
```
ditto

lukechampine commented on plugins/Files/js/components/uploaddialog.js:
```diff
@@ -18,7 +18,10 @@ const UploadDialog = ({source, path, actions}) => {
-				<div>Would you like to upload {source.length} {source.length === 1 ? 'item' : 'items'}?</div>
+				<div>
+					Would you like to upload {source.length}{' '}
+					{source.length === 1 ? 'item' : 'items'}?
```
ditto

lukechampine commented on plugins/Files/js/components/usagestats.js:
```diff
@@ -2,30 +2,70 @@ import PropTypes from 'prop-types'
+					</li>
+					<li className="download-spending-breakdown">
+						{' '}
+						Download Spending: {downloadspending} SC{' '}
```
these too. I guess there's gonna be a lot. Bummer.

lukechampine commented:
this mostly looks fine to me. It's a bit aggressive on splitting up long lines (which traditionally we don't care about so much), and I also dislike indenting `case` blocks in `switch` statements; is the formatter configurable? I imagine we'd want to add it to our package.json somehow for easy invocation?

eddiewang commented:
The formatter is configurable - the scripts exist in `package.json` under `npm run format` as well as some git hooks on the larger PR, but for this one I just manually ran the script. I'm currently working on some failing tests that still exist on master. Once that's done, I'd like to rebase this PR so that Travis builds work properly again, so hold off on merging this for now.

eddiewang committed:
update lint configs

eddiewang committed:
Fix babelrc

eddiewang committed:
remove unnecessary {' '}

eddiewang commented:
@lukechampine I removed the unnecessary whitespacing. You are right. Lots of whitespace that was accidentally being left in the HTML. But that's also why I prefer using {' '} for whitespacing in JSX - it makes it really clear that you have spaces in text.

The default width for long lines in Standard JS is 80 chars. I'm inclined to stick with the defaults (look at how clean the eslintrc is compared to the old one) for now, just because it's a well-known standard that people are used to. It's super easy to change that in the future, we just add a custom rule and run `npm run format`.

 I also updated the babelrc so that webpack builds properly, and so that tests passed based on my fix from yesterday's PR.

I've rebased from master and this PR should be good to go.

DavidVorick commented:
Is it able to distinguish between text lines and code lines?

Not a big deal / blocking factor for me either way. I would probably target 80 chars for text lines (since readability of a broken up line is not impacted) and 160 chars for code lines (since readability takes a huge hit if you split up a code line - unless you do it very elegantly, which I don't imagine a script is able to do.

eddiewang committed:
remove eslintrc
